### PR TITLE
Added Vive Pro controller mapping support

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

Adds proper support for the Vive Pro in the generic open vr controller mapping

## Target of the change:

Is this enhancement for:

- Core (core framework, interfaces and definitions)
- Platform (new or updated platform)

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/OpenVR/GenericOpenVRController.cs [hunk 1]
```diff
@@ -37,102 +37,104 @@ public GenericOpenVRController(TrackingState trackingState, Handedness controlle
...
+            new MixedRealityInteractionMapping(9, "Unity Button Id 0", AxisType.Digital, DeviceInputType.ButtonPress, KeyCode.JoystickButton0),
             // Oculus Remote - Button.Two Press
             // Oculus Touch Controller - Button.Two Press
             // Valve Knuckles Controller - Right Controller Outer Face Button
-            new MixedRealityInteractionMapping(9, "Unity Button Id 1", AxisType.Digital, DeviceInputType.ButtonPress, KeyCode.JoystickButton1),
-            new MixedRealityInteractionMapping(10, "WMR Touchpad Touch", AxisType.Digital, DeviceInputType.TouchpadTouch, KeyCode.JoystickButton19),
-            new MixedRealityInteractionMapping(11, "WMR Touchpad Position", AxisType.DualAxis, DeviceInputType.Touchpad, ControllerMappingLibrary.AXIS_19, ControllerMappingLibrary.AXIS_20),
+            new MixedRealityInteractionMapping(10, "Unity Button Id 1", AxisType.Digital, DeviceInputType.ButtonPress, KeyCode.JoystickButton1),
+            new MixedRealityInteractionMapping(11, "WMR Touchpad Touch", AxisType.Digital, DeviceInputType.TouchpadTouch, KeyCode.JoystickButton19),
```
Comment (SimonDarksideJ) [inline XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/OpenVR/GenericOpenVRController.cs]: There is no WMR definition in OpenVR, can we update the description on these items to reflect that please.

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/OpenVR/GenericOpenVRController.cs]: I was only adding the grip to fulfill the changes of the PR.

Can we remove it in a subsequent PR?

**Comment by StephenHodgson** [comment]
> I see you have added a spatial grip, even though the OpenVR spec does not have a "spatial grip" definition (and neither to vive wands)

Do either of us have a device to test this with?

File: XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/OpenVR/GenericOpenVRController.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/OpenVR/GenericOpenVRController.cs]: I see, so you are saying those definitions were already there. Then yes, we can remove that in another PR.

**Comment by SimonDarksideJ** [comment]
You know me, I seem to have all the devices, as the (un)"Offical" tester :D 

I'll get this tested later today by hooking up the Vive

**Pull request merged**